### PR TITLE
Add cancel billing agreement function

### PR DESCRIPTION
**Pull request by rohanr-bolt**
#### What does this PR do?
Adds function to allow users to cancel billing agreements with the id and a note (reason)
#### Where should the reviewer start?
The function is just hitting the cancel endpoint with an id
#### How should this be manually tested?
The function is hitting the endpoint with an id, there is a unit test as well
#### Any background context you want to provide?
follow api at https://developer.paypal.com/docs/limited-release/reference-transactions/#cancel-billing-agreement

### CODE REVIEW

**Review by Jankyboy — DISMISSED** [review]

File: billing_agreements.go
Comment (thomasf) [inline billing_agreements.go]: you probably meant CancelPaymentsBillingAgreement ?

**Comment by rohanr-bolt** [comment]
@plutov @roopakv looks like this needs approval by someone with write access, can one of you look at this by chance

**Review by thomasf — COMMENTED** [review]
> @plutov @roopakv looks like this needs approval by someone with write access, can one of you look at this by chance

You still havent replied to my comment.

`CancelPaypalBillingAgreement` has a completley redundant "paypal" in it. This whole package is paypal and you don't need to mention it in methods. 

`CancelPaymentBillingAgreement` is contextually correct for the paypal api.

If there are other methods that use this clearly wrong naming convetions, create the correct method names for them and leave the old ones with a ` // Deprecated: use SomeMethodName instead.`. 

Please don't introduce more pointless and obviously bad naming.

**Comment by rohanr-bolt** [comment]
> > @plutov @roopakv looks like this needs approval by someone with write access, can one of you look at this by chance
> 
> You still havent replied to my comment.
> 
> `CancelPaypalBillingAgreement` has a completley redundant "paypal" in it. This whole package is paypal and you don't need to mention it in methods.
> 
> `CancelPaymentBillingAgreement` is contextually correct for the paypal api.
> 
> If there are other methods that use this clearly wrong naming convetions, create the correct method names for them and leave the old ones with a ` // Deprecated: use SomeMethodName instead.`.
> 
> Please don't introduce more pointless and obviously bad naming.

I don't see any previous comment from you on my side but I will fix

**Comment by rohanr-bolt** [comment]
@plutov  any chance this can get a look

**Review by roopakv — APPROVED** [review]

**Pull request merged**
